File: core/src/main/java/com/github/albertosh/flitetrakr/input/AppInput.java
package com.github.albertosh.flitetrakr.input;

import java.util.Scanner;

class AppInput implements IAppInput {

    private Scanner scanner;

    public AppInput(Scanner scanner) {
        this.scanner = scanner;
    }

    @Override
    public boolean hasNext() {
        return scanner.hasNext();
    }

    @Override
    public String nextLine() {
        return scanner.nextLine();
    }

    @Override
    public void close() {
        scanner.close();
    }


}

File: core_kotlin/src/main/kotlin/com/github/albertosh/flitetrakr/input/AppInput.kt
package com.github.albertosh.flitetrakr.input

import com.github.albertosh.flitetrakr.util.language.LanguageUtils
import com.github.albertosh.flitetrakr.util.language.Message
import java.io.Closeable
import java.io.File
import java.io.FileNotFoundException
import java.util.*

interface IAppInput : Closeable {

    fun hasNext(): Boolean

    fun nextLine(): String

}

open class AppInput(private val scanner: Scanner) : IAppInput {

    companion object {
        fun setupInput(args: Array<String>): IAppInput {
            if (args.size > 0)
                try {
                    return FileInput(args[0])
                } catch (e: FileNotFoundException) {
                    throw IllegalStateException(LanguageUtils.getMessage(Message.INPUT_FILE_NOT_FOUND_ERROR))
                }
            else
                return StandardInput()
        }

    }

    override fun hasNext(): Boolean {
        return scanner.hasNext()
    }

    override fun nextLine(): String {
        return scanner.nextLine()
    }

    override fun close() {
        scanner.close()
    }

}

class FileInput(file: String) : AppInput(Scanner(File(file)))

class StandardInput() : AppInput(Scanner(System.`in`))
